terraform/sponsored_bucket.tf

```diff
@@ -158,3 +159,39 @@ data "aws_iam_policy_document" "sponsored_bucket" {
     ]
   }
 }
+
+module "sponsored_embargo_bucket" {
+  source              = "./modules/dandiset_bucket"
+  bucket_name         = "dandiarchive-embargo"
+  versioning          = false
+  log_bucket_name     = "dandiarchive-embargo-logs"
+  log_bucket_owner_id = data.aws_canonical_user_id.sponsored_account.id
+  policy_json         = data.aws_iam_policy_document.sponsored_embargo_bucket.json
+  providers = {
+    aws = aws.sponsored
+  }
+}
+
+data "aws_iam_policy_document" "sponsored_embargo_bucket" {
```

Comment:
I would expect that the essential / duplicative parts of this policy would be defined within the `dandiset_bucket` module.

The module can output the name / ID of the bucket, which can be used to attach supplemental policies outside with `aws_s3_bucket_policy`.

Reply:
https://github.com/hashicorp/terraform/issues/10543 suggests that it's only possible to attach a single policy to a bucket. You can have a single policy with multiple statements, but I'm not aware of a way to attach an additional statement to the module. Am I missing something?

Reply:
I moved the Heroku IAM user access to a user policy. Any customized access rules can now be bound as the bucket policy.